geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/LayerReaderFactory.scala

```diff
@@ -57,26 +55,17 @@ abstract class FilteringLayerReaderWrapper()
 
   def read(
     keyType: String,
-    valueType: String,
     layerName: String,
     zoom: Int
   ): (JavaRDD[Array[Byte]], String) = {
     val id = LayerId(layerName, zoom)
 
-    (keyType, valueType) match {
-      case ("SpatialKey", "Tile") => {
-        val results = layerReader.read[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](id)
-        PythonTranslator.toPython(results)
-      }
-      case ("SpatialKey", "MultibandTile") => {
+    keyType match {
+      case "SpatialKey" => {
         val results = layerReader.read[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](id)
```

Comment:
We still need to decide if we're doing `Tile` or `MultibandTile` query for geotrellis because the Avro schema used in storage is different. So what has to happen here is `val header = attributeStore. readHeader[SpatialKey]` and inspect `valueClass` String. If the layer stores `Tile` we must query for tile and then wrap them in `ArrayMultibandTile` to return to python. Since this has to happen at every read/query it should be made generic enough to be a function.